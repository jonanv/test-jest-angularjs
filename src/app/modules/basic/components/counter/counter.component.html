<h1>Counter {{ counter }}</h1>
<hr>

<div class="d-flex gap-2">
  <button
    class="btn btn-primary"
    (click)="increseBy();">
    +1
  </button>
  <button
    class="btn btn-primary"
    (click)="decreaseBy();">
    -1
  </button>
</div>
