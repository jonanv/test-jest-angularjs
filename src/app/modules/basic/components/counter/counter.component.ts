import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-counter',
  templateUrl: './counter.component.html'
})
export class CounterComponent implements OnInit {

  public counter: number = 0;

  constructor() { }

  public ngOnInit(): void {
  }

  public increseBy(): void {
    this.counter++;
  }

  public decreaseBy(): void {
    this.counter--;
  }

}
