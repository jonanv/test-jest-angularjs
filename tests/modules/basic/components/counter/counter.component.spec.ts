import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../../../src/app/modules/basic/components/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CounterComponent]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement as HTMLElement;
  });

  test('Debe crearse el componente', () => {
    expect(component).toBeTruthy();
  });

  test('Debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('Debe de incrementar basado en el argumento (5)', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(5);
  });

  test('Debe de incrementar y decrementar en 1 al presionar los botones', () => {
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    expect(component.counter).toBe(1);

    buttons[1].click();
    buttons[1].click();
    expect(component.counter).toBe(-1);
  });

  test('Cambiar el counter debe de actualizar la etiqueta h1', () => {
    component.increaseBy(10);
    fixture.detectChanges();

    const h1 = compiled.querySelector('h1');
    expect(h1?.textContent).toContain('10');
  });

});
