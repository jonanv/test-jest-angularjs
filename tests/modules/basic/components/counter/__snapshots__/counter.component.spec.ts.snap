// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent Debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <h1>
    Counter 0
  </h1>
  <hr />
  <div
    class="d-flex gap-2"
  >
    <button
      class="btn btn-primary"
    >
       +1 
    </button>
    <button
      class="btn btn-primary"
    >
       -1 
    </button>
  </div>
</div>
`;
